### PR TITLE
Feature/back 1282 update charts tim

### PR DESCRIPTION
**Pull request by ttschampel**

### CODE REVIEW

File: scp/charts/configuration/templates/server/vsvc.yaml
Comment (ttschampel) [inline scp/charts/configuration/templates/server/vsvc.yaml:1]: This should be removed

Comment (c-r33d) [reply scp/charts/configuration/templates/server/vsvc.yaml:1]: Was that a recent commit to the add-config-to-helm-repo?

Comment (ttschampel) [reply scp/charts/configuration/templates/server/vsvc.yaml:1]: It wasn't in the add-config-to-helm-repo branch.  I think it was left over from attempting to use istio to route traffic vs. using the virtru-proxy

File: gateway/templates/istio-ingress/vsvc.yaml [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+{{- if .Values.gatewayModes.outboundDlp.enabled }}
+apiVersion: networking.istio.io/v1alpha3
+kind: VirtualService
+metadata:
+  name: gateway
+spec:
+  gateways:
+    - {{ include "gateway.name" . }}-outbound-dlp
+  hosts:
+    - '*' # Should be changed with the hostname of the mail server.
```
Comment (ttschampel) [inline gateway/templates/istio-ingress/vsvc.yaml:11]: Is this something that is known ahead of time? If, so lets template it as a value

**Review by ttschampel — COMMENTED** [review]
Just the one question on mail hostname config

File: gateway/templates/istio-ingress/vsvc.yaml (hunk 1 above)
Comment (c-r33d) [reply gateway/templates/istio-ingress/vsvc.yaml]: I believe the Istio Gateway should be the FQDN of the domain, and the destination host should be the name of the service.

**Review by strantalis — COMMENTED** [review]
The readme needs to be updated with the new variables. It would be better to think about using something like helm docs in the future.

File: gateway/templates/configmap-base.yaml [hunk 2]
```diff
@@ -21,6 +21,13 @@ data:
   GATEWAY_ENCRYPTION_KEY_PROVIDER: {{ .Values.additionalConfig.cks.keyProvider }}
   GATEWAY_CKS_SESSION_KEY_EXPIRY_IN_MINS: {{ .Values.additionalConfig.cks.sessionKeyExpiry | quote }}
   GATEWAY_DLP_CACHE_DURATION: {{ .Values.additionalConfig.dlpRuleCache | quote }}
+  {{- if eq .Values.additionalConfig.abac.enabled true }}
+  OIDC_CLIENT_ID: {{ .Values.additionalConfig.abac.oidcClientId | quote }}
+  OIDC_TOKEN_URL: {{ .Values.additionalConfig.abac.oidcTokenUrl | quote }}
+  TAGGING_PDP_LOCATION: {{ .Values.additionalConfig.abac.taggingPdpLocation | quote }}
```
[the remark this answers is not in the document]
Comment (c-r33d) [reply gateway/templates/configmap-base.yaml:27]: Yes, but without the protocol

File: gateway/values.yaml [hunk 3]
```diff
@@ -134,6 +135,7 @@ appSecrets:
       <dkim-public-key>
     privateKey: |
       <dkim-private-key>
+  oidcClientSecret: <oidc-client-secret> # Used for making reqs to an OIDC provider to obtain an access token. Specifically when making reqs to access-pep.
```
[the remark this answers is not in the document]
Comment (strantalis) [reply gateway/values.yaml:138]: @c-r33d Will get a Jira ticket updated to track the effort to add ability to reference an existing kubernetes secret.

File: gateway/values.yaml [hunk 4]
```diff
@@ -176,4 +178,23 @@ additionalConfig:
...
+    trimRecipients: false
+
+istioIngress:
+  enabled: false
+  # Use an existing istio gateway
+  existingGateway:
+  # Name of istio gateway selector
+  gatewaySelectors: 
+    istio: ingress
+  ingressHostname:
```
Comment (strantalis) [inline gateway/values.yaml:198]:
```suggestion
  ingressHostnames: 
```

File: gateway/templates/istio-ingress/gateway.yaml [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+spec:
+  selector:
+    {{- toYaml .Values.istioIngress.gatewaySelectors | nindent 12 }}
+  servers:
+    - port:
+        number: {{ .Values.gatewayModes.outboundDlp.port }}
+        name: gateway
+        protocol: TCP
+      hosts:
+        {{- toYaml .Values.istioIngress.ingressHostname | nindent 12 }}
```
Comment (strantalis) [inline gateway/templates/istio-ingress/gateway.yaml:16]:
```suggestion
        {{- toYaml .Values.istioIngress.ingressHostnames | nindent 12 }}
```

File: gateway/templates/istio-ingress/vsvc.yaml [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+{{- if .Values.gatewayModes.outboundDlp.enabled }}
+apiVersion: networking.istio.io/v1beta1
+kind: VirtualService
+metadata:
+  name: gateway
+spec:
+  gateways:
+    - {{ include "gateway.name" . }}-outbound-dlp
+  hosts:
+    {{- toYaml .Values.istioIngress.ingressHostname | nindent 12 }}
```
Comment (strantalis) [inline gateway/templates/istio-ingress/vsvc.yaml:11]:
```suggestion
    {{- toYaml .Values.istioIngress.ingressHostnames | nindent 12 }}
```

File: gateway/templates/istio-ingress/vsvc.yaml [hunk 7]
```diff
@@ -0,0 +1,21 @@
+{{- if .Values.istioIngress.enabled }}
+{{- if .Values.gatewayModes.outboundDlp.enabled }}
+apiVersion: networking.istio.io/v1beta1
+kind: VirtualService
+metadata:
+  name: gateway
```
Comment (strantalis) [inline gateway/templates/istio-ingress/vsvc.yaml:6]: Do you want to append `outbound-dlp` to the name here as well.

**Comment by strantalis** [comment]
@c-r33d LGTM

**Pull request merged**